### PR TITLE
feat: Add WAL meta endpoint client

### LINKED ISSUE #724: Add API to see what segments exist in the WAL along with the writer IDs and min and max sequence numbers
Located at:

```
GET /iox/api/v1/databases/:name/wal/meta
```

Query parameters: `limit=10` default to 10 segments. Or `newer_than=2021-02-02T12:000` optional, only return segments with a created at time greater than or equal to the passed in time.

Response:

```
{
  "segments" : [
    {
        "location" : "1/mydb/wal/000/000/002.wal",
        "size" : 23432,
        "created_at": "2021-02-02T12:00:00Z",
        "writers": [
            {"id" : 1, "min_sequence": 23, "max_sequence": 59}
        ],
        "persisted" : true
  ],
}
```

For segments that haven't been persisted, their location will be where the file gets written to when it gets written. The created_at time will also be a placeholder for when this segment was opened, but will ultimately get set by the object store on creation of the file.

Add this to the `influxdb_iox_client` create.

### PR DESCRIPTION
**Pull request by tustvold**
Builds on top of #768 

Closes #724

### CODE REVIEW

File: influxdb_iox_client/src/client.rs [hunk 1]
```diff
@@ -262,6 +292,44 @@ mod tests {
...
+        let r = c
+            .get_wal_metadata(&db_name, &Default::default())
+            .await
+            .expect("get metadata failed");
+
+        assert_eq!(r.segments.len(), 1);
+        assert_eq!(r.segments[0].size, 0);
+        assert_eq!(r.segments[0].writers.len(), 0);
+        assert!(r.segments[0].persisted.is_none());
+    }
```
Comment (alamb) [inline influxdb_iox_client/src/client.rs:331]: I suggest also adding a test here for at least one of the error cases ( Requesting a non-existent database, or a database not with a wal) to ensure the error plumbing is wired up correctly

**Review by alamb — APPROVED** [review]
Looks good to me 

I ran the tests locally via
```
TEST_IOX_ENDPOINT=http://localhost:8080 cargo test -p influxdb_iox_client
```

And everything passed and looks great. Nice work

**Review by e-dard — APPROVED** [review]
NICE. LGTM 👍

**Comment by tustvold** [comment]
There is an issue with the HTTP API as it stands, with it returning a blank payload for 404 responses. This prevent the WalNotFound error from propagating correctly. As this API is slated to be replaced, will close this ticket for now

**Pull request closed without merging**